02-peer-protocol.md

```diff
@@ -430,6 +446,515 @@ would create a Denial of Service risk; therefore, forgetting it is recommended
+    * [`u16`:`num_inputs`]
+    * [`u16`:`num_outputs`]
+
+
+##### Requirements
+
+Both nodes:
+  - MUST send this message after `accept_channel2` has been sent/received.
+
+The sending node:
+  - MUST ensure that the `num_inputs` corresponds to the total sum of all `num_inputs`
+    sent in all `funding_add_input` messages that originated from them
+  - MUST ensure that the `num_outputs` corresponds to the total sum of all `num_outputs`
+    sent in all `funding_add_output` messages that originated from them
+
+The receiving node:
+  - MUST fail the channel if:
+    - the `num_inputs` does not correspond to the total sum of all `num_inputs`
+      received in all `funding_add_input` messages
+    - the `num_outputs` does not correspond to the total sum of all `num_outputs`
+      received in all `funding_add_output` messages
+    - the total satoshis of the senders inputs is less than their outputs plus
+      the funding_sats, specified earlier
+  - if is the `opener`:
+    - MAY fail the channel if:
+      - the fee cost of the proposed funding transaction is deemed exorbitant.
+  - if has not yet sent a `funding_add_complete`:
+    - MUST send its `funding_add_complete` message.
+  - otherwise:
+    - MUST use the sent and received `input_info` and `output_info`
+      to create the funding transaction, using `max_witness_len`
+      for each `input_info` and `feerate_per_kw_funding` as specified in
+      `open_channel2`.
+    - MUST send `commitment_signed`.
+
+
+##### Rationale
+
+Each node must have a complete set of the transaction inputs and outputs,
+to derive the funding transaction and subsequent commitment signatures.
+
+`funding_add_complete` concludes a node's transmission of its contributions
+to the funding transaction, which serve as an early check that all
+`funding_add_input` and `funding_add_output` messages have been received.
+
+Upon successful exchange of `funding_add_complete` messages, both nodes
+should build the funding transaction and update their channel id
+to be transmitted in `commitment_signed`.
+
+
+### The `commitment_signed` Message
+
+This message is sent by the opening node. It contains the signatures for
+the first commitment transaction.
+
+Rationale and Requirements are the same as listed above,
+for [`commitment_signed`](#commiting-updates-so-far-commitment_signed) with the following additions.
+
+#### Requirements
+
+The sending node:
+  - MUST send zero HTLC's.
+
+The receiving node:
+  - if the message has one or more HTLC's:
+    - MUST fail the channel.
+  - if it has not already transmittted its `commitment_signed`:
+    - MUST send `commitment_signed`
+  - Otherwise:
+    - MUST send `funding_signed2`
+
+#### Rationale
+
+The first commitment transaction has no HTLC's in it.
+
+Note that the `commitment_signed` message will include the `channel_id` derived from
+the `funding_txid`, instead of the `temporary_node_id`.
+
+
+### The `funding_signed2` Message
```

Comment:
Message name is a bit confusing, funding_signedv1 refers to accepter giving commitment tx signatures to fundee, here it refers to signatures exchange for the funding tx (even now wondering if a funding_signed distinct from commitment_signed wasn't a spec oversight..)